src/server/scripts/World/go_scripts.cpp

```diff
@@ -821,19 +821,21 @@ class go_midsummer_music : public GameObjectScript
                             if (!IsHolidayActive(HOLIDAY_FIRE_FESTIVAL))
                                 break;
 
-                            Map::PlayerList const& players = me->GetMap()->GetPlayers();
-                            for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+                            std::list<Player*> targets;
+                            Acore::AnyPlayerInObjectRangeCheck check(me, me->GetVisibilityRange(), false);
+                            Acore::PlayerListSearcherWithSharedVision<Acore::AnyPlayerInObjectRangeCheck> searcher(me, targets, check);
+                            Cell::VisitWorldObjects(me, searcher, me->GetVisibilityRange());
+                            for (std::list<Player*>::const_iterator iter = targets.begin(); iter != targets.end(); ++iter)
```

Comment:
```suggestion
                            for (Player const* player : targets)
```

Maybe? :)

Reply:
That is much improvement! Haha. Thank you!